solana/bridge-escrow/programs/bridge-escrow/src/lib.rs

```diff
@@ -318,20 +343,11 @@ pub mod bridge_escrow {
         // Handle Native SOL Transfer
         if intent.token_out == System::id().to_string() {    
             // Perform SOL transfer from Solver to User
-            let ix = solana_program::system_instruction::transfer(
-                solver.key,
-                &intent.user_in,
-                amount_out,
-            );
-    
-            invoke(
-                &ix,
-                &[
-                    solver.to_account_info(),
-                    accounts.user_token_out_account.to_account_info(),
-                    accounts.system_program.to_account_info(),
-                ],
-            )?;
+            let out_user_account = accounts.user_account.clone().unwrap();
+            
+            **solver.lamports.borrow_mut() -= amount_out;
```

Comment:
since solver account is owned by system program, u need to call system program to transfer just like how you do it in `escrow_and_store_intent` method.